_episodes/07-vector-data-in-python.md

```diff
@@ -149,126 +165,221 @@ array([109222.03325 , 469461.512625, 140295.122125, 510939.997875])
-how to use spatial data attributes stored in our data to plot
-different features.
+~~~
+fields_wells = fields.sjoin(wells[['bro_id', 'geometry']])
+print(fields_wells.shape)
+print(fields_wells.columns)
+~~~
+{: .language-python}
 
+~~~
+(79, 8)
+Index(['category', 'gewas', 'gewascode', 'jaar', 'status', 'geometry',
+       'index_right', 'bro_id'],
+      dtype='object') 
+~~~
+{: .output}
 
+This will result in a `GeodataFrame` of all polygons intersecting the wells, adding the well IDs as a new column: `bro_id`.
 
-> ## Challenge: Import Line and Point Vector Datasets
->
-> Using the steps above, load the waterways and groundwater well vector datasets (`data/status_vaarweg.zip` and
-> `data/brogmwvolledigeset.zip`, respectively) into Python using `geopandas`. Name your variables `waterways_nl` and
-> `wells_nl` respectively.
->
-> Answer the following questions:
->
-> 1. What type of spatial features (points, lines, polygons) are present in each dataset?
->
-> 2. What is the CRS and total extent (bounds) of each dataset?
->
-> 4. How many spatial features are present in each file?
+> ## Challenge: count the number of field polygons within a distance from the points
+> Use the `sjoin` function to join two polygons: the field polygons (`fields`, read in from `data/fields_cropped.shp`) and the wells 500m buffer polygons (`wells_cx_500mbuffer` from the previous exercise). And answer the following questions: 
+> 
+> - How many fields are intersecting the 500m buffer of wells?
 >
+> - How many wells are intersecting the fields with a 500 buffer?
+> 
 > > ## Answers
-> >
-> > First we import the datasets:
-> > ```python
-> > waterways_nl = gpd.read_file("data/status_vaarweg.zip")
-> > wells_nl = gpd.read_file("data/brogmwvolledigeset.zip")
-> > ```
-> >
-> > Then we check the types:
-> > ```python
-> > waterways_nl.type
-> > ```
-> >
-> > ```python
-> > wells_nl.type
-> > ```
-> > We also check the CRS and extent of each object:
-> > ```python
-> > print(waterways_nl.crs)
-> > print(waterways_nl.total_bounds)
-> > print(wells_nl.crs)
-> > print(wells_nl.total_bounds)
-> > ```
-> > To see the number of features in each file, we can print the dataset objects in a Jupyter notebook or use the `len()` function.
-> > `waterways_nl` contains 91 lines and `wells_nl` contains 51664 points.
+> > ~~~
+> > fields_wells_buffer = fields.sjoin(wells_cx_500mbuffer)
+> > print(fields_wells_buffer.shape)
+> > print(fields_wells_buffer.index.unique().shape) # Take the unique of the index
+> > print(fields_wells_buffer['bro_id'].unique().shape) # Take the unique of the bro_ids
+> > 
+> > # validation
+> > print(fields.clip(wells_cx_500mbuffer).shape)
+> > ~~~
+> > {: .language-python}
+> > Note that a polygon can fall in the 500m buffer of multiple points. So we should use `.unique()` function to get the actual number of the polygons.
 > {: .solution}
 {: .challenge}
 
+## (optional) Modify the geometry of a GeoDataFrame
```

Comment:
Maybe just leave out the optional and every instructor can decide whether this is relevant or not for the audience?

```suggestion
## Modify the geometry of a GeoDataFrame
```

Reply:
Good suggestion, I will add it to the instrutors note instead.